src/python/bot/tasks/corpus_pruning_task.py

```diff
@@ -840,6 +873,16 @@ def _try_save_coverage_information():
         'Failed to save corpus pruning result: %s.' % repr(e))
 
 
+def choose_cross_pollination_strategy(current_fuzzer_name):
+  """Chooses cross pollination strategy. In seperate function to mock for
+    predictable test behaviror."""
+  # Do not choose tagged method if there is no tag for the current fuzzer.
+  if corpus_tagging.get_similarly_tagged_fuzzers(current_fuzzer_name):
```

Comment:
this calls into Datastore, let's not always do it unconditionally. Instead, do the `random.choice` first, if it returns random then you just return random; then check if the tagged method is possible, and branch the result based on it